src/OpenSage.Game/Scene3D.cs

```diff
@@ -157,6 +157,9 @@ public void SetPlayers(IEnumerable<Player> players, Player localPlayer)
 
             LocalPlayer = localPlayer;
 
+            var MainUnit = LocalPlayer.SelectedUnits.First();
```

Comment:
Do the local players always have selected units at the start? A player isn't required to have starting units at all when configured like that inside the PlayerTemplate.ini. 
This would fail when the SelectedUnits count is 0, so please but an if statement arround the two lines.